lib/6to5/transformation/transformers/es6-let-scoping.js

```diff
@@ -112,14 +112,40 @@ LetScoping.prototype.run = function () {
   }
 };
 
+function replace(node, parent, scope, context, remaps) {
+  if (!t.isReferencedIdentifier(node, parent)) return;
+
+  var remap = remaps[node.name];
+  if (!remap) return;
+
+  var own = scope.get(node.name, true);
+  if (own === remap.node) {
+    node.name = remap.uid;
+  } else {
+    // scope already has it's own declaration that doesn't
+    // match the one we have a stored replacement for
+    if (context) context.skip();
+  }
+}
+
+var replaceTraverser = {
+  enter: replace
+};
+
+function traverseReplace(node, parent, scope, remaps) {
+  replace(node, parent, scope, null, remaps);
+  traverse(node, replaceTraverser, scope, remaps);
+}
+
 /**
  * Description
  */
 
 LetScoping.prototype.remap = function () {
-  var letRefs = this.letReferences;
-  var scope   = this.scope;
-  var file    = this.file;
+  var hasRemaps = false;
```

Comment:
I merged `master` into `3.0.0` after this pull request so I'm not sure why it hasn't been updated.